components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaFetchRecords.java

```diff
@@ -244,6 +244,9 @@ protected void createConsumer() {
             // this may throw an exception if something is wrong with kafka consumer
             this.consumer = kafkaConsumer.getEndpoint().getKafkaClientFactory().getConsumer(kafkaProps);
 
+            var krbLocation = kafkaConsumer.getEndpoint().getConfiguration().getKerberosConfigLocation();
+            System.setProperty("java.security.krb5.conf", krbLocation);
```

Comment:
What happens if this option is not set, then the value is null

Reply:
@davsclaus Yes correct, missed that point. Thanks!
As there is no default location, can we update the code like this?
var krbLocation = kafkaConsumer.getEndpoint().getConfiguration().getKerberosConfigLocation();
if (krbLocation != null) {
          System.setProperty("java.security.krb5.conf", krbLocation);
}

Reply:
Okay let us know when you have fixed this

Reply:
Fixed this issue. Thanks!